### DISCUSSION TITLE
Contribution with DongshanPi PicoW

### DISCUSSION THREAD

**Discussion by Vexisu** [linux-chenxing/linux-chenxing.org#90] [General]
Hi guys,
from some time I have DongshanPi PicoW board on hand. I want to contribute in some way to this project. If you have some time, You could guide me with what I can do to somehow help. Also, I'm kinda noob in terms of Linux kernel, so I want to learn something way through.

**Comment by fifteenhex** [comment]
Hi,

I guess the first thing to do is get mainline going...
I started putting together a buildroot setup for the picow and perdita (my board, same chip).

https://github.com/breadbee/buildroot_perdita

I haven't touched it in a while because I got busy. Maybe we could start on getting that running on your board, fix up any issues you faced getting it working and go from there.

**Reply by Vexisu** [reply]
Okay. Somehow managed to build from the buildroot, but I had to fix Meson deps (had to change branches from `head` to `master`). Also had to remove games from the build because of paranoid path checking. The worst part is that I don't know how to boot without the ethernet plugged in lmao

**Reply by fifteenhex** [reply]
>The worst part is that I don't know how to boot without the ethernet plugged in lmao

I think the easiest thing to do is use the u-boot DFU mode to upload the images to the flash.
I have notes that I'll add to the readme once I finish work today.

**Reply by Vexisu** [reply]
Have you uploaded the instructions? I can't find them anywhere.
